src/fontra/views/editor/visualization-layer-definitions.js

```diff
@@ -281,6 +279,138 @@ registerVisualizationLayerDefinition({
   },
 });
 
+registerVisualizationLayerDefinition({
+  identifier: "fontra.guidelines",
+  name: "Guidelines",
+  selectionMode: "editing",
+  userSwitchable: true,
+  defaultOn: false,
+  zIndex: 500,
+  screenParameters: {
+    fontSize: 10,
+    strokeWidth: 1,
+    strokeLength: 1000,
+    originMarkerRadius: 4,
+    strokeDash: 3,
+    margin: 5,
+    iconSize: 12,
+  },
+  colors: {
+    strokeColor: "#0006",
+    boxColor: "#FFFB",
+    color: "#000",
+  },
+  colorsDarkMode: {
+    strokeColor: "#FFF8",
+    boxColor: "#1118",
+    color: "#FFF",
+  },
+  draw: (context, positionedGlyph, parameters, model, controller) => {
+    context.font = `${parameters.fontSize}px fontra-ui-regular, sans-serif`;
+    context.textAlign = "center";
+
+    // Draw glyph guidelines
+    for (const guideline of positionedGlyph.glyph.guidelines) {
+      _drawGuideline(context, parameters, guideline);
+    }
+
+    // TODO: Font Guidelines
+  },
+});
+
+function _drawGuideline(context, parameters, guideline) {
+  context.save();
```

Comment:
- Please use withSavedState() instead of context.save()/context.restore(). That is a better pattern, and is used all over the place. They can nest as well.
- Please clean up the transformation logic, so that you don't need to "rotate back" to draw the lock icon.

The overall structure will be something like this:
```
    withSavedState(() => {
      translate to guideline origin
      draw lock icon or the "node"
      withSavedState(() => {
        rotate
        scale 1, -1
        draw name
        draw lines
      }  
    }
```

Reply:
Thanks a lot for the great and valid feedback. Done.